vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/registry/impl/FSPackageRegistry.java

```diff
@@ -0,0 +1,528 @@
+        dispatch(PackageEvent.Type.REMOVE, id, null);
+    }
+
+    @Nonnull
+    @Override
+    public Set<PackageId> packages() throws IOException {
+        Set<PackageId> packageIds = new HashSet<>();
+        
+
+        Collection<File> files = FileUtils.listFiles(getHomeDir(), META_SUFFIXES, true);
+        for (File file : files) {
+            InstallState state = getInstallState(file);
+            PackageId id = state.getPackageId();
+            if (id != null) {
+                packageIds.add(id);
+            }
+        }
+        return packageIds;
+    }
+
+    /**
+     * Returns the path of this package.this also includes the version, but
+     * never the extension (.zip).
+     *
+     * @param id the package id
+     * @return the path of this package
+     */
+    public String getInstallationPath(PackageId id) {
+        return getRelativeInstallationPath(id);
+    }
+
+    @Override
+    public void installPackage(Session session, RegisteredPackage pkg, ImportOptions opts, boolean extract) throws IOException, PackageException {
+
+        if (session == null){
+            String msg = "Installation of packages only supported when session is set.";
+            log.error(msg);
+            throw new PackageException(msg);
+        }
+        
+        // For now FS based persistence only supports extraction but no reversible installation
+        if (!extract) {
+            String msg = "Only extraction supported by FS based registry";
+            log.error(msg);
+            throw new PackageException(msg);
+        }
+        VaultPackage vltPkg = pkg.getPackage();
+        if (vltPkg instanceof ZipVaultPackage) {
+            try {
+                ((ZipVaultPackage)vltPkg).extract(session, opts);
+            } catch (RepositoryException e) {
+                throw new IOException(e);
+            }
+            FSPackageStatus targetStatus = extract ? FSPackageStatus.EXTRACTED : FSPackageStatus.INSTALLED;
+            updateInstallState(vltPkg.getId(), targetStatus);
+        }
+    }
+
+    @Override
+    public void uninstallPackage(Session session, RegisteredPackage pkg, ImportOptions opts) throws IOException, PackageException {
+        String msg = "Uninstallation not supported by FS based registry";
+        log.error(msg);
+        throw new PackageException(msg);
+    }
+    
+    private void updateInstallState(PackageId pid, FSPackageStatus targetStatus) throws IOException {
+        InstallState state = getInstallState(pid);
+        setInstallState(pid, targetStatus, state.getFilePath(), state.isExternal());
+    }
+
+    private void setInstallState(PackageId pid, FSPackageStatus targetStatus, String filePath, boolean external) throws IOException {
+        File metaData = getPackageMetaDataFile(pid);
+        
+        if (targetStatus.equals(FSPackageStatus.NOTREGISTERED)) {
+            metaData.delete();
+        } else {
+            OutputStream out = FileUtils.openOutputStream(metaData);
```

Comment:
are you sure you want to use XML? isn't json or properties much easier? (and potentially faster) ?

Reply:
@tripodsan  reason for xml is just that I wanted to reuse the same serialization  in vault-core already used for execution plan and didn#t want to introduce yet another serializer